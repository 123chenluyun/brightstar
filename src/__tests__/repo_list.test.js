import React from 'react';
import ReactDOM from 'react-dom';
import RepoList from '../components/repo_list.js';
import { mount,shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('<RepoList />', () => {

  let props, repo_list;

  beforeEach(() => {
    props = {
        repos: [{id:1, contributors_url:"www.test.com",  stargazers_count: 100, name: "test", description: "test description" }]
    };
    repo_list = mount(<RepoList {...props}/> );   
  });  

  it('renders correctly', () => {
    const repo_list = renderer.create(
     <RepoList {...props}/>
    ).toJSON();
    expect(repo_list).toMatchSnapshot();
  });

  it('allows us to set props', () => {
      expect(repo_list.props()).toEqual(props);
  });

  it('has proper classes and elements', () => {
      expect(repo_list.find('.col').length).toEqual(1);
      expect(repo_list.find('.table .table-stripped .table-hover').length).toEqual(1);  
      expect(repo_list.find('.stargazers-th').text()).toEqual("Stargazers");  
      expect(repo_list.find('.name-th').text()).toEqual("Name");  
      expect(repo_list.find('.contributors-th').text()).toEqual("Top Contributors");  
  });


});