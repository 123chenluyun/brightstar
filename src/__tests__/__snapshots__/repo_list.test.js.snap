// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoList /> renders correctly 1`] = `
<div
  className="col"
>
  <table
    className="table table-stripped table-hover"
  >
    <tbody>
      <tr>
        <th
          className="stargazers-th"
        >
          Stargazers
        </th>
        <th
          className="name-th"
        >
          Name
        </th>
        <th
          className="contributors-th"
        >
          Top Contributors
        </th>
      </tr>
      <tr>
        <td
          className="star-cell"
        >
          <svg
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m37.5 15l-12.2-1.6-5.3-10.9-5.3 10.9-12.2 1.6 9 8.2-2.3 11.8 10.8-5.8 10.8 5.8-2.3-11.8 9-8.2z"
              />
            </g>
          </svg>
          100
        </td>
        <td
          className="repo-details-cell"
        >
          <h4>
            test
          </h4>
          test description
        </td>
        <td
          className="contributors-cell"
        />
      </tr>
    </tbody>
  </table>
</div>
`;
